modules/library/src/main/scala/zio/elasticsearch/query/ElasticIntervalQuery.scala

```diff
@@ -0,0 +1,356 @@
+package zio.elasticsearch.query
+
+import zio.elasticsearch.ElasticPrimitive.ElasticPrimitiveOps
+import zio.elasticsearch.query.options.{HasAnalyzer, HasUseField}
+import zio.json.ast.Json
+import zio.json.ast.Json.{Arr, Obj, Str}
+import zio.{Chunk, NonEmptyChunk}
+import zio.elasticsearch.Field
+
+sealed trait BoundType
+sealed trait Inclusive extends BoundType
+sealed trait Exclusive extends BoundType
+
+case object InclusiveBound extends Inclusive
+case object ExclusiveBound extends Exclusive
+
+final case class Bound[B <: BoundType](value: String, boundType: B)
```

Comment:
Can we reuse classes from the Elastic range query? If not, let's consider what we can keep private.